components/brave_rewards/core/engine/util/brotli_util.h

```diff
@@ -9,12 +9,12 @@
 #include <string>
 #include <string_view>
 
+#include "base/containers/span.h"
+
 namespace brave_rewards::internal {
 namespace util {
 
-bool DecodeBrotliString(std::string_view input,
-                        size_t uncompressed_size,
-                        std::string* output);
+bool DecodeBrotliString(std::string_view input, base::span<uint8_t> output);
```

Comment:
Is it possible to remove the `reinterpret_cast`s without changing the function signature? It feels inconsistent to change the signature of this function but not `DecodeBrotliStringWithBuffer`.